src/App.jsx

```diff
@@ -14,33 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, sortField, reversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case 'alpha':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseOrder, setReverseOrder] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverseOrder,
+  );
+
+  const handleReverseClick = () => {
+    setReverseOrder(!reverseOrder);
+  };
+
+  const handleResetClick = () => {
+    setSortField('');
+    setReverseOrder(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortField('alpha');
+          }}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== 'alpha',
+          })}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => {
+            setSortField('length');
+          }}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== 'length',
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            handleReverseClick();
+          }}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': reverseOrder === false,
+          })}
+        >
+          Reverse
+        </button>
+
+        {visibleGoods[0] !== goodsFromServer[0] ? (
```

Comment:
Not the best way to check if array is modified; use your control variables `sortField` and `reverseOrder` for that